### PR TITLE
net-mail/mboxgrep: Fix call to undeclared library function strcmp

### PR DESCRIPTION
**Pull request by listout**
Closes: https://bugs.gentoo.org/897938

### CODE REVIEW

**Review by juippis — COMMENTED** [review]
Compiles fine, but i still get:
```
 * QA Notice: Found the following implicit function declarations in configure logs:
 *   /var/tmp/portage/net-mail/mboxgrep-0.7.9-r3/work/mboxgrep-0.7.9/config.log:175 - exit
 * Check that no features were accidentally disabled.
 * See https://wiki.gentoo.org/wiki/Modern_C_porting.
```

**Comment by listout** [comment]
@juippis The warning is coming from the `configure` file (line no. 2560) as the example test code does not have `#include <stdlib.h>`. If I try adding patch/diff fixing that to the origin patch, I'm getting the following warning:

```
>> scrub-patch files/mboxgrep-0.7.9-musl-missing-strcmp.patch
Successfully scrubbed:
 - files/mboxgrep-0.7.9-musl-missing-strcmp.patch (13.90% reduction)

QA: files/mboxgrep-0.7.9-musl-missing-strcmp.patch has generated warnings:

013: +++ b/configure
^^^ if this 'configure' is autotools-generated, it's preferable to patch
    configure.ac and use eautoreconf instead

If patch is intended for the ::gentoo tree, would be good to review.
Note that these checks are tentative and cannot consider everything.
https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/index.html
```

**Comment by juippis** [comment]
> @juippis The warning is coming from the `configure` file (line no. 2560) as the example test code does not have `#include <stdlib.h>`. If I try adding patch/diff fixing that to the origin patch, I'm getting the following warning:
> 
> ```
> >> scrub-patch files/mboxgrep-0.7.9-musl-missing-strcmp.patch
> Successfully scrubbed:
>  - files/mboxgrep-0.7.9-musl-missing-strcmp.patch (13.90% reduction)
> 
> QA: files/mboxgrep-0.7.9-musl-missing-strcmp.patch has generated warnings:
> 
> 013: +++ b/configure
> ^^^ if this 'configure' is autotools-generated, it's preferable to patch
>     configure.ac and use eautoreconf instead
> 
> If patch is intended for the ::gentoo tree, would be good to review.
> Note that these checks are tentative and cannot consider everything.
> https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/index.html
> ```

Looking at the sources, it does ship **configure.ac** file. So could you try by updating your patch, inheriting *autotools.eclass*, and adding `src_prepare()` phase with 
```
	default
	eautoreconf
```
that should work.

**Comment by juippis** [comment]
(and while at it, bump EAPI to latest available please!)

File: net-mail/mboxgrep/mboxgrep-0.7.9-r3.ebuild [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+)
+
+src_prepare() {
+	default
+	eautoreconf
+}
+
+src_configure() {
+	econf \
+		$(use_with dmalloc no yes)
```
Comment (juippis) [inline net-mail/mboxgrep/mboxgrep-0.7.9-r3.ebuild:39]: Hmm I don't remember if this syntax changed between EAPIs 6 and 8, but I'm getting:
```
 * QA Notice: Unrecognized configure options:
 * 
 * 	--without-no
 * 	--without-no
```
`./configure --prefix=/usr --build=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu --mandir=/usr/share/man --infodir=/usr/share/info --datadir=/usr/share --sysconfdir=/etc --localstatedir=/var/lib --datarootdir=/usr/share --docdir=/usr/share/doc/        mboxgrep-0.7.9-r3 --htmldir=/usr/share/doc/mboxgrep-0.7.9-r3/html --libdir=/usr/lib64 --without-no`

Something like
```suggestion
	$(use_with dmalloc dmalloc $(usex dmalloc yes no))
```
should work though.

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**